datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -779,17 +779,21 @@ impl ExecutionPlan for SortExec {
 
         debug!("End SortExec's input.execute for partition: {}", partition);
 
-        let result = do_sort(
-            input,
-            partition,
-            self.expr.clone(),
-            self.metrics_set.clone(),
-            context,
-        )
-        .await;
-
-        debug!("End SortExec::execute for partition {}", partition);
-        result
+        Ok(Box::pin(RecordBatchBoxStream::new(
+            self.schema(),
+            futures::stream::once(
+                do_sort(
```

Comment:
A nicer fix would be to make ExternalSort stream, or less async, or some combination of the two, but in the short term this is a quick fix